pkg/exporter/process.go

```diff
@@ -204,6 +207,16 @@ func (ep *ExportingProcess) CloseConnToCollector() {
 	}
 }
 
+// CheckConnToCollector checks whether the connection from exporter is still open
+// by trying to read from connection. Closed connection will return EOF from read.
+func (ep *ExportingProcess) CheckConnToCollector() bool {
```

Comment:
I see that we are piggybacking on the export goroutine in Antrea. This adds some overhead (including some delay) in the exporter goroutine. Prefer to handle this in a separate goroutine in the library. You probably want to avoid locking on the client connection.

Did you get a chance to look at this connection struct from net library: https://pkg.go.dev/net#TCPConn.SetKeepAlive ?

Reply:
I have considered using `setKeepAlive`. In that case we will need to keep a `TCPConn` for tcp and `net.Conn` for udp instead of `net.Conn` in our collecting process and handle write and close differently for tcp and udp connections. It does not have read overhead or blocking the connection. 
Using current check will have multiple reads to one byte memory. If we use goroutine for checking the connection, it will not block the connection too.
These two solutions look similar to me. Do you have any preference?

Reply:
I see. In the first option, there is quite a bit of refactoring needed. The benefit of this option is utilizing the standardized library code that takes care of all scenarios.
I am ok with the second option too. As the read is done in a separate goroutine, the overhead is not so much. If we are using the lock on the connection, we might be blocking the connection while doing either read or write. Isn't that correct? For reading on the connection, we can set a deadline so that we will not block for a long time. 
Any thoughts on the periodic interval for this goroutine?

Reply:
Got it. Haven't thought about using lock to prevent concurrent write and read. In that case, I think refactoring to TCPConn will be more convenient since we don't need to handle the locking and clean up for goroutine (also good for encapsulate the checking for connection openness). Just had a draft test on refactoring. Will also test at Antrea side to see whether it will work well.

Reply:
Toke a closer look and try on refactoring  to `TCPConn`, it does not work well with tls settings and will have multiple retries before return. Using `Read` can directly return the result and we can customize the periodic interval. I re-write the way of checking so that not much change will be needed in Antrea. Please take a look Thanks!